packages/core/src/locales/ua.ts

```diff
@@ -0,0 +1,126 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; verb: string }> = {
+  string: { unit: "символів", verb: "матиме" },
+  file: { unit: "байтів", verb: "матиме" },
+  array: { unit: "елементів", verb: "матиме" },
+  set: { unit: "елементів", verb: "матиме" },
+};
+
+function getSizing(origin: string): { unit: string; verb: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "NaN" : "число";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "масив";
+      }
+      if (data === null) {
+        return "null";
+      }
+
+      if (Object.getPrototypeOf(data) !== Object.prototype && data.constructor) {
+        return data.constructor.name;
+      }
+    }
+  }
+  return t;
+};
+
+const Nouns: {
+  [k in $ZodStringFormats | (string & {})]?: string;
+} = {
+  regex: "вхідні дані",
+  email: "адреса електронної пошти",
+  url: "URL",
+  emoji: "емодзі",
+  uuid: "UUID",
+  uuidv4: "UUIDv4",
+  uuidv6: "UUIDv6",
+  nanoid: "nanoid",
+  guid: "GUID",
+  cuid: "cuid",
+  cuid2: "cuid2",
+  ulid: "ULID",
+  xid: "XID",
+  ksuid: "KSUID",
+  datetime: "дата та час ISO",
+  date: "дата ISO",
+  time: "час ISO",
+  duration: "тривалість ISO",
+  ipv4: "адреса IPv4",
+  ipv6: "адреса IPv6",
+  cidrv4: "діапазон IPv4",
+  cidrv6: "діапазон IPv6",
+  base64: "рядок у кодуванні base64",
+  base64url: "рядок у кодуванні base64url",
+  json_string: "рядок JSON",
+  e164: "номер E.164",
+  jwt: "JWT",
+  template_literal: "вхідні дані",
+};
+
+const error: errors.$ZodErrorMap = (issue) => {
+  switch (issue.code) {
+    case "invalid_type":
+      return `Неправильні вхідні дані: очікується ${issue.expected}, отримано ${parsedType(issue.input)}`;
+    // return `Неправильні вхідні дані: очікується ${issue.expected}, отримано ${util.getParsedType(issue.input)}`;
```

Comment:
Do we need this? 😅

Reply:
> Do we need this commented out code? =)

<img width="771" alt="image" src="https://github.com/user-attachments/assets/fee5f077-304c-4413-8f62-8136fdf0b2de" />
I just followed the en.ts file, which also has this commented line of code